File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/operations/downloadtracks.py
import dataclasses
from typing import Optional
from ..shared import security as shared_security
from ..shared import url as shared_url


@dataclasses.dataclass
class DownloadTracksPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class DownloadTracksSecurity:
    customer_access_code: shared_security.SchemeCustomerAccessCode = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclasses.dataclass
class DownloadTracksRequest:
    path_params: DownloadTracksPathParams = dataclasses.field()
    security: DownloadTracksSecurity = dataclasses.field()
    

@dataclasses.dataclass
class DownloadTracksResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    url: Optional[shared_url.URL] = dataclasses.field(default=None)
    

File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/operations/downloadtracks.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DownloadTracksPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}


export class DownloadTracksSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  customerAccessCode: shared.SchemeCustomerAccessCode;
}


export class DownloadTracksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DownloadTracksPathParams;

  @SpeakeasyMetadata()
  security: DownloadTracksSecurity;
}


export class DownloadTracksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  url?: shared.Url;
}
